kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -121,4 +125,22 @@ public Path getLogPath() {
     public Path getDataPath() {
         return dataPath;
     }
+
+    public long getTimestamp(Engine engine) {
```

Comment:
Please add a docstring.

Reply:
Also, it looks like the Commit file be read every time `getTimestamp` is invoked. This is inefficient.
1. We should read it once and store it in a variable. 
2. As a future optimization, this should be read as part of `LogReplay.loadTableProtocolAndMetadata` and then passed by SnapshotManager to Snapshot

Reply:
I think it will be only read once? `inCommitTimestampOpt` will be present for the following reads except the first read.